File: demo/person.py
class Person:
    name: str
    age: int

    def __init__(self, name: str, age: int) -> None:
        self.name = name
        self.age = age

    def greet(self) -> None:
        print(f"Hello, I am {self.name}")


if __name__ == "__main__":
    name: str = input("Enter name: ")
    age: int = int(input("Enter age: "))
    p: Person = Person(name, age)
    p.greet()
    print(p.age)

File: demo/person.rs
#[allow(unused_imports)]
use std::collections::HashMap;
use std::io::{self, Write};

fn __read_input(prompt: &str) -> String {
    print!("{}", prompt);
    io::stdout().flush().ok();
    let mut input = String::new();
    io::stdin().read_line(&mut input).ok();
    input.trim().to_string()
}

#[derive(Debug, Clone)]
struct Person {
    pub name: String,
    pub age: i64,
}

impl Person {
    pub fn new(name: String, age: i64) -> Self {
        Self {
            name: name,
            age: age,
        }
    }
    pub fn greet(&mut self) {
        println!("{}", format!("Hello, I am {}", self.name));
    }
}

fn main() {
    let mut name: String = __read_input("Enter name: ");
    let mut age: i64 = __read_input("Enter age: ").parse::<i64>().unwrap_or(0);
    let mut p: Person = Person { name: name, age: age };
    p.greet();
    println!("{}", p.age);
}
